pkg/catalogv2/helm/helm3.go

```diff
@@ -8,15 +8,28 @@ import (
 	"io/ioutil"
 	"strings"
 
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-
 	v1 "github.com/rancher/rancher/pkg/apis/catalog.cattle.io/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
 
 	"helm.sh/helm/v3/pkg/release"
 )
 
-func isHelm3(labels map[string]string) bool {
-	return labels["owner"] == "helm"
+// isHelm3 checks if the object is a helm3 release secret ?
+// obj can be configmap or secret. Helmv3 only creates secrets.
+// So, we first check if it is a secret or not and then check
+// for Secret Type. The Secret Type should contain string "helm.sh/release.v1"
+// https://github.com/helm/helm/blob/main/pkg/storage/driver/secrets.go#L231
+func isHelm3(obj runtime.Object) bool {
+	secret, ok := obj.(*corev1.Secret)
```

Comment:
@aiyengar2 The second [commit](https://github.com/rancher/rancher/pull/40693/commits/e529244c4b4cb550ceac766be376eb0e24dbb8a1) resolves the error irrespective of the resouce type whether it is a secret of a configmap.

isHelm3() function should check if the secret is of helm3 or not. The correct way to check this is not by checking the labels since any resource can have that label and need not be a helm3-created resource. 

Helm source code identifies a helm3 secret by adding the secretType as commented here https://github.com/helm/helm/blob/main/pkg/storage/driver/secrets.go#L231. Also, helm3 only deals with secrets, and helm2 deals with configmap and secrets. A configmap cannot be created by helm3 as stated in the docs.

Reply:
FYI to my knowledge, Helm 3 does support ConfigMaps; it’s just not widely used.

See https://helm.sh/docs/topics/advanced/#configmap-storage-backend

Also in the code, https://github.com/helm/helm/blob/a499b4b179307c267bdf3ec49b880e3dbd2a5591/pkg/storage/driver/cfgmaps.go#L39

Reply:
Hmm..Okay. thanks for pointing this out. I will remove the first commit and continue with the label filter. 

Continue with the second commit.